### DISCUSSION TITLE
Please add range aggregations to visualizations

### DISCUSSION THREAD

**Discussion by tehowe** [grafana/grafana#55645] [Dashboards, Panels & Visualizations]
It appears that Grafana would benefit from a range aggregation capability similar to that offered by Kibana in graphs like the following:

![Screenshot from 2022-09-22 16-37-28](https://user-images.githubusercontent.com/60449561/191846530-4bad80bf-41ae-47d6-b10f-cfdcffe7e0f9.png)

The code for that would look something like this in Kibana, if that helps any:

`{
  "aggs": {
    "2": {
      "range": {
        "field": "score",
        "ranges": [
          {
            "from": null,
            "to": -6
          },
          {
            "from": -6,
            "to": -4
          },
          {
            "from": -4
          }
        ],
        "keyed": true
      },
      "aggs": {
        "3": {
          "date_histogram": {
            "field": "@timestamp",
            "interval": "3h",
            "time_zone": "America/New_York",
            "min_doc_count": 1
          }
        }
      }
    }
  },
  "size": 0, ETC`

The closest thing I found to be possible in Grafana was a term value grouping like the following, but that only allows two buckets per given term range - counting from the top and counting from the bottom.

![Screenshot from 2022-09-22 16-44-30](https://user-images.githubusercontent.com/60449561/191849397-10788a8c-73d8-4ada-b94d-55d1aec01ef4.png)

There seems to be some call for this - it's certainly important to us to be able to have time series and stacked histograms based on ranges within a term. I found this request as well https://community.grafana.com/t/range-aggregations/1909/3 but couldn't find anything submitted here under "range aggregation", "split term", and similar. Thanks!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
